src/calendarworker.cpp

```diff
@@ -784,7 +780,7 @@ void CalendarWorker::loadData(const QList<CalendarData::Range> &ranges,
     foreach (const QString &id, instanceList)
         mStorage->loadIncidenceInstance(id);
 
-    if (!ranges.isEmpty()) {
+    if (!ranges.isEmpty() && reset) {
```

Comment:
Trying this with my calendar that has a bunch of repeating events per week, it doesn't seem to load them. Didn't go investigating more than removing this addition gets them back, but at least the mResetPending on manager is initialized false.

Reply:
You're right, I didn't pay enough attention. Actually, I've some excluded notebooks, so the `mResetPending` was set to `true` on start up in my case when the excluded notebook setting is read.

I think it's safe to initialise `mResetPending` to `true` to get the first initial load to run without other side effects. I've amended the commit in that sense.

Reply:
Hm, should this isEmpty() check be here actually? Thinking what happens if e.g. notebooks get loaded on startup, excluded notebooks trigger reset/loadData from the manager and this triggers without ranges set for some reason.